queries.sql

```diff
@@ -7,4 +7,43 @@ SELECT name, escape_attempts from animals WHERE weight_kg >10.5;
 SELECT date_of_birth FROM animals WHERE name = 'Agumon' OR name='Pikachu';
 SELECT * FROM animals WHERE neutered=true;
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
-SELECT * FROM animals WHERE name NOT IN('Gabumon');
\ No newline at end of file
+SELECT * FROM animals WHERE name NOT IN('Gabumon');
+
+BEGIN;
+UPDATE animals SET species='unspecified';
+SELECT * FROM ANIMALS ;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+UPDATE animals SET species='digimon'
+WHERE name LIKE '%mon';
+UPDATE animals SET species='pokemon'
+WHERE species is NULL;
+SELECT * FROM animals;
+COMMIT;
+SELECT * FROM animals;
+BEGIN;
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+SAVEPOINT delete1;
+DELETE FROM animals WHERE date_of_birth > 'jan-1-2022';
+UPDATE animals SET weight_kg = weight_kg*-1;
+ROLLBACK TO delete1;
+UPDATE animals SET weight_kg = weight_kg*-1
+WHERE weight_kg <0;
+SELECT * FROM animals; 
+COMMIT;
```

Comment:
- [OPTIONAL] The `SAVEPOINT` statement should be after this statement:

```sql
DELETE FROM animals WHERE date_of_birth > 'jan-1-2022';
``` 
The statement above will delete the entry for the animal named _**Ditto**_. It is after this statement that you should have the SAVEPOINT statement. This is the reason why most of your screenshots have the wrong results. I would advise you to implement this change because it is a crucial point that affects most of the results of most queries in this milestone and the next.

#### Hint:

```sql
BEGIN;
DELETE FROM animals WHERE date_of_birth > 'jan-1-2022';
SAVEPOINT delete1;
UPDATE animals SET weight_kg = weight_kg*-1;
ROLLBACK TO delete1;
UPDATE animals SET weight_kg = weight_kg*-1
WHERE weight_kg <0;
SELECT * FROM animals; 
COMMIT;
```